### PR TITLE
3.2.3 list of group maslova

### PR DESCRIPTION
**Pull request by maslovakat**

### CODE REVIEW

**Review by andrew-kysel — APPROVED** [review]

File: front/src/common/redux/actions/groupListAction.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { CURRENT_GROUP } from './types';
+
+export const setCurrentGroup = (data) => {
+  return {
+    type: CURRENT_GROUP,
```
Comment (Alan18081) [inline front/src/common/redux/actions/groupListAction.js]: It is better to use a verb for the name of this action type

File: front/src/common/redux/reducers/groupListReducer.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+import * as types from '../actions/types';
+
+const initialState = {
+  currentGroup: 1,
```
Comment (Alan18081) [inline front/src/common/redux/reducers/groupListReducer.js]: It is better to store in `state` actual group object with all of the data, not the id itself

File: front/src/components/list-of-group/GroupListPage.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { Layout, Menu } from 'antd';
+import 'antd/dist/antd.css';
+import { AddGroupButton, SiderStyle } from './styledListOfGroup';
+import { setCurrentGroup } from '../../common/redux/actions/groupListAction';
+import { connect } from 'react-redux';
+
+const { Content } = Layout;
+
+class GroupListPage extends Component {
+    onHandleGroup = (data) => {
```
Comment (Alan18081) [inline front/src/components/list-of-group/GroupListPage.js]: On `componentDidMount` you should send a request and get all of the groups that are available for the logged user.

File: front/src/components/list-of-group/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import GroupListPage from './GroupListPage';
```
Comment (Alan18081) [inline front/src/components/list-of-group/index.js]: Why do we need `GroupListPage.js` at all? Please, move all the code from `GroupListPage.js` to the index.js inside list-of-group` folder

File: front/src/components/list-of-group/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+
+export default GroupListPage;
```
Comment (Alan18081) [inline front/src/components/list-of-group/index.js]: Component name and name of folder in which it is stored should match

File: front/src/common/redux/actions/groupListAction.js (hunk 1 above)
Comment (maslovakat) [reply front/src/common/redux/actions/groupListAction.js]: done

File: front/src/common/redux/reducers/groupListReducer.js (hunk 2 above)
Comment (maslovakat) [reply front/src/common/redux/reducers/groupListReducer.js]: done

File: front/src/components/list-of-group/GroupListPage.js (hunk 3 above)
Comment (maslovakat) [reply front/src/components/list-of-group/GroupListPage.js]: done

File: front/src/components/list-of-group/index.js (hunk 4 above)
Comment (maslovakat) [reply front/src/components/list-of-group/index.js]: done

File: front/src/components/list-of-group/index.js (hunk 5 above)
Comment (maslovakat) [reply front/src/components/list-of-group/index.js]: done

File: front/src/common/validators/form.validator.js [hunk 6]
```diff
@@ -28,21 +28,21 @@ export function validatePhoneNumber(_, value) {
   }
 }
 export function validateEmail(_, value) {
-    const emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  const emailRegex = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
```
Comment (so2niko) [inline front/src/common/validators/form.validator.js:31]: Don`t understand the reason of this editing

**Review by so2niko — APPROVED** [review]
Not a problem, but interesting about this regExp

**Review by staspitsyk — APPROVED** [review]

File: front/src/pages/login/index.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    );
+  }
+}
+
+const mapStateToProps = ({ login: { errorMessage, isLoggedIn } }) => ({
+  errorMessage,
+  isLoggedIn
+});
+
+const mapDispatchToProps = (dispatch) => {
```
Comment (Alan18081) [inline front/src/pages/login/index.js]: Please, simplify it to the form of `mapStateToProps`

File: front/src/common/validators/form.validator.js (hunk 6 above)
Comment (maslovakat) [reply front/src/common/validators/form.validator.js:31]: with this backslash I had warnings with React

File: front/src/pages/login/index.js (hunk 7 above)
Comment (maslovakat) [reply front/src/pages/login/index.js]: done

**Review by antonpanchenk0 — APPROVED** [review]

**Pull request merged**